### ISSUE TITLE
Works without VPN; doesn't work with VPN

### ISSUE THREAD

**Issue by SturmB** [MarkusMcNugen/docker-qBittorrentvpn#16]
Caveat: I am new to the VPN world.

I've been able to get this docker image up and running with `VPN_ENABLED` set to `no`. For me, that's a major accomplishment, as I am also brand-new to the Docker world. I then decided to take advantage of the VPN capabilities of the image, so I purchased a subscription with TorGuard.

I downloaded two sets of .ovpn files that they provide (one for TCP, one for UDP) and placed one of the UDP files into the instructed folder, `/config/openvpn`. While the container does seem to start, I cannot log in to qBittorrent's web app. Checking the logs, I am greeted with these as the last two lines:

```
ERROR: Cannot open TUN/TAP dev /dev/net/tun: No such file or directory (errno=2)
Exiting due to fatal error
```

I am not sure that I should post the entire log, as it might contain some personal info. However, this particular error does seem to be common enough that a cursory Google search comes up with many solutions. Unfortunately, none seem to be applicable to this particular Docker container.

For reference, I am running this on a Synology DiskStation running DSM 6.2.1. I installed Docker from the Package Manager and was able to install this image and set all of its environment variables and such via the Docker GUI on DSM.

I think this is most likely _not_ a problem with the docker image, but with setting up the VPN properly. Can anyone please help me to fix this issue?

**Comment by Philipose** [comment]
Just had this issue. Realized I forgot to add the "privileged: true" line to my docker-compose file. Adding the privilege parameter should fix it.

**Comment by globalnomad** [comment]
> Just had this issue. Realized I forgot to add the "privileged: true" line to my docker-compose file. Adding the privilege parameter should fix it.

@Philipose Is it possible to set "privileged: true" in environment variables when setting this up via a Synology Docker wizard? I'm getting this error as well.

**Comment by Philipose** [comment]
Personally I don't know because I haven't tried Synology's implementation. I did miss the paragraph saying it was on DSM.
It looks like with the correct version, you may be able to.
See links:

https://forum.synology.com/enu/viewtopic.php?t=118103

https://www.synology.com/en-global/knowledgebase/DSM/help/Docker/docker_dsm
(About 3 lines down talking about high privilege) 

Again I cant be sure but hopefully that helps

**Comment by globalnomad** [comment]
@Philipose Thanks!

**Comment by BlernsAreLoaded** [comment]
I'm having this same issue but I've ticked the "Execute container using high privilege" box. Same results with that option ticked and unticked on two different Synology NAS devices. When checking out the filesystem of the container it doesn't appear to have those folders mentioned. Anyone seen this issue before? Was there another step required to make it work that I'm missing?

**Comment by globalnomad** [comment]
@BlernsAreLoaded So I ended up deciding not to use this container and went with a transmission+ovpn container. That said, you may be able to follow similar instructions [here](https://www.reddit.com/r/VPNTorrents/comments/900w78/xpost_rsynology_setting_up_the/) regarding SSH and the TUN.sh file creation and execution. Note that I had an issue creating the file on windows and then loading it onto the Synology with differences in endline types - I had to convert them to Unix endlines using Sublime Text (or Notepad++ etc.). See [this comment](https://www.reddit.com/r/VPNTorrents/comments/900w78/xpost_rsynology_setting_up_the/e6ycxzp/) in the the thread I linked above.

**Comment by SturmB** [comment]
It's been a while since I posted this issue and since then, I've moved to a different Docker image: [arch-qbittorrentvpn](https://hub.docker.com/r/binhex/arch-qbittorrentvpn/).

I did run into this same "TUN/TAP" issue with that Docker image, however, so I sought help on a Synology community Discord. They pointed me to [this article](https://ruimarinho.github.io/post/fix-tun-tap-not-available-on-a-synology-nas/), which solved my problem.

So, because I am not using this Docker image anymore, plus finding a solution with a similar Docker image, I will go ahead and close this issue.
